-- 404: Not Found























































































































































































































































































































































































































































































































































































































































































































































































































































































































































































































--[[
       
            ░▒▓███████▓▒░░▒▓██████▓▒░░▒▓█▓▒░░▒▓█▓▒░░▒▓██████▓▒░  
           ░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░ 
           ░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░ 
            ░▒▓██████▓▒░░▒▓█▓▒░░▒▓█▓▒░▒▓████████▓▒░▒▓█▓▒░░▒▓█▓▒░ 
                  ░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░ 
                  ░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░ 
           ░▒▓███████▓▒░ ░▒▓██████▓▒░░▒▓█▓▒░░▒▓█▓▒░░▒▓██████▓▒░  
                                                     
                         𝑆𝑂𝐻𝑂 𝑈𝐼 𝑎 𝑠𝑐𝑟𝑖𝑝𝑡 ℎ𝑢𝑏.
                  𝐴 𝑝𝑟𝑜𝑗𝑒𝑐𝑡 𝑝𝑟𝑒𝑣𝑖𝑜𝑢𝑠𝑙𝑦 𝑏𝑦 𝑀𝑦𝑡ℎ𝑖𝑐 𝑇𝑒𝑎𝑚.
       𝑆𝑖𝑛𝑐𝑒 𝑀𝑦𝑡ℎ𝑖𝑐 𝑤𝑎𝑠 𝑜𝑓𝑓𝑖𝑐𝑖𝑎𝑙𝑙𝑦 𝑑𝑖𝑠𝑐𝑜𝑛𝑡𝑖𝑛𝑢𝑒𝑑. 02/22/2025 (mm/DD/yyyy)
]]--

local HttpService = game:GetService("HttpService")
local StarterGui = game:GetService("StarterGui")

local PLACE_ID = game.PlaceId
local BASE_URL = "https://raw.githubusercontent.com/OdaGenie/SOHO-UI/refs/heads/main/games"

local function notify(title, text)
	pcall(function()
		StarterGui:SetCore("SendNotification", {
			Title = title,
			Text = text,
			Duration = 5
		})
	end)
end

local function loadGameScript()
	local url = BASE_URL .. PLACE_ID .. ".luau"

	local success, response = pcall(function()
		return game:HttpGet(url)
	end)

	if not success or not response or response:find("404") then
		return notify("SOHO UI", "This game is not supported.")
	end

	local ok, err = pcall(function()
		loadstring(response)()
	end)

	if not ok then
		warn("Script error:", err)

		notify(
			"Script Error",
			"Failed to load game script."
		)
	end
end

loadGameScript()
